### ISSUE TITLE
Rule proposal: `no-empty-array-push`

### ISSUE THREAD

**Issue by Zamiell** [sindresorhus/eslint-plugin-unicorn#1382]
I just found the following bug in some of my TypeScript code:

```ts
myArray.push();
```

This was a typo, and I had actually intended to write:

```ts
myArray.push(foo);
```

I can't imagine that a JavaScript/TypeScript programmer would **ever** intend to call the `push` method with 0 arguments.
Thus, it seems to me that this would always be a bug.
Subsequently, some kind of linter rule should automatically catch this bug.

<br />

## Fail

```js
myArray.push();
```

## Pass

```js
myArray.push(1);
```

```js
// Even though the array is empty, this still does not count as an "empty" array push,
// because we are still explicitly providing at least one argument
const anotherArray = [];
myArray.push(...anotherArray);
```

**Comment by sindresorhus** [comment]
I think https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1356 would solve this too.

**Comment by Zamiell** [comment]
Today I realized that this pitfall applies to **all** variadic functions, not just `Array.push`.
So, I made the rule: https://github.com/IsaacScript/isaacscript/blob/main/packages/eslint-plugin-isaacscript/docs/rules/require-variadic-function-argument.md

Feel free to copy paste it into unicorn if you want, as I think it is a really good rule.
